TenureInformationApi.Tests/V1/UseCase/EditTenureDetailsUseCaseTests.cs

```diff
@@ -72,5 +80,76 @@ public async Task EditTenureDetailsUseCaseWhenTenureExistsReturnsTenureResponseO
             response.EndOfTenureDate.Should().Be(gatewayResponse.UpdatedEntity.EndOfTenureDate);
             response.TenureType.Code.Should().Be(gatewayResponse.UpdatedEntity.TenureType.Code);
         }
+
+        [Fact]
+        public async Task EditTenureDetailsUseCaseWhenNoChangesSNSGatewayIsntCalled()
+        {
+            var mockQuery = _fixture.Create<TenureQueryRequest>();
+            var mockRequestObject = _fixture.Create<EditTenureDetailsRequestObject>();
+            var mockRawBody = "";
+            var mockToken = _fixture.Create<Token>();
+
+            // setup mock gateway to return UpdateEntityResult with no changes
+            var gatewayResult = MockUpdateEntityResultWhereNoChangesAreMade();
+
+            _mockGateway
+                .Setup(x => x.EditTenureDetails(It.IsAny<TenureQueryRequest>(), It.IsAny<EditTenureDetailsRequestObject>(), It.IsAny<string>()))
+                .ReturnsAsync(gatewayResult);
+
+            // call usecase method
+            var response = await _classUnderTest.ExecuteAsync(mockQuery, mockRequestObject, mockRawBody, mockToken).ConfigureAwait(false);
+
+            // assert result is TenureResponseObject
+            response.Should().BeOfType(typeof(TenureResponseObject));
+
+            // assert that sns factory wasnt called
+            _tenureSnsGateway.Verify(x => x.Publish(It.IsAny<TenureSns>(), It.IsAny<string>(), It.IsAny<string>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task EditTenureDetailsUseCaseWhenChangesSNSGatewayIsCalled()
+        {
+            var mockQuery = _fixture.Create<TenureQueryRequest>();
+            var mockRequestObject = _fixture.Create<EditTenureDetailsRequestObject>();
+            var mockRawBody = "";
+            var mockToken = _fixture.Create<Token>();
+
+            // setup mock gateway to return UpdateEntityResult with no changes
+            var gatewayResult = MockUpdateEntityResultWhereChangesAreMade();
+
+            _mockGateway
+                .Setup(x => x.EditTenureDetails(It.IsAny<TenureQueryRequest>(), It.IsAny<EditTenureDetailsRequestObject>(), It.IsAny<string>()))
+                .ReturnsAsync(gatewayResult);
+
+            // call usecase method
+            var response = await _classUnderTest.ExecuteAsync(mockQuery, mockRequestObject, mockRawBody, mockToken).ConfigureAwait(false);
+
+            // assert result is TenureResponseObject
+            response.Should().BeOfType(typeof(TenureResponseObject));
+
+            // assert that sns factory wasnt called
+            _tenureSnsGateway.Verify(x => x.Publish(It.IsAny<TenureSns>(), It.IsAny<string>(), It.IsAny<string>()), Times.Once);
```

Comment:
You could validate the contents of the TenureSns object here, and that the 2nd parameter is the correct topic name.